### PR TITLE
webcore: free Blob's owned content type on drop

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
Blob kept its content type as a raw slice pointer plus an ownership flag, with no drop glue. Consumers that drop a duplicated Blob by value (the file read handler, `PathOrBlob`, `BlobOrStringOrBuffer`, `AnyBlob`, the object-URL registry) relied on an explicit `deinit()` that several of them never call, so the heap copy made by `dupe_with_content_type` was stranded on every `Bun.file(p, { type }).text()` and similar paths.

This adds `impl Drop for Blob` that frees an owned content type, makes `borrowed_view()` return a self-owning `dupe()` instead of aliasing the source's allocation (which would double-free under the new Drop), and hoists the deep copy in `dupe_with_content_type` ahead of construction. The remaining `Blob { .., ..Default::default() }` literals are spelled out since functional-record-update is rejected for Drop types.

Adds three tests to `test/js/web/fetch/blob.test.ts`: an RSS regression test for repeated `.text()` on a file handle with a large custom type, and two behavioral tests confirming the source handle's type survives reads and `Bun.write`. Existing content-type ownership tests in that file still pass.

### CODE REVIEW

File: test/js/web/fetch/blob.test.ts [hunk 1]
```diff
@@ -325,6 +325,76 @@ test("dupe() preserves allocated content_type for Body clone", () => {
...
+      // the measurement reflects live memory on ASAN builds too.
+      ASAN_OPTIONS: [bunEnv.ASAN_OPTIONS, "quarantine_size_mb=0"].filter(Boolean).join(":"),
+    },
+    stdout: "pipe",
+    stderr: "pipe",
+  });
+  const [stdout, stderr, exitCode] = await Promise.all([proc.stdout.text(), proc.stderr.text(), proc.exited]);
+  const { deltaMiB } = JSON.parse(stdout);
+  expect(deltaMiB).toBeLessThan(isASAN ? 400 : 40);
+  expect(exitCode).toBe(0);
```
[the remark this answers is not in the document]
Comment (Jarred-Sumner) [reply test/js/web/fetch/blob.test.ts:354]: Done in 54c240cace5f — `expect(stderr).toBe("")` now runs before `JSON.parse(stdout)`.

[the remark this answers is not in the document]
Comment (Jarred-Sumner) [reply test/js/web/fetch/blob.test.ts:354]: Done in 54c240cace5f — (1) added `expect(stderr).toBe("")` before parsing stdout; (2) dropped the `quarantine_size_mb=0` override and its comment entirely and noted that LeakSanitizer (via the exit-code assertion) is the regression guard on ASAN builds, matching how the other `isASAN ? 400` leak tests in the repo treat the ASAN RSS number.

File: src/runtime/node/types.rs [hunk 2]
```diff
@@ -1984,12 +1984,11 @@ impl PathOrBlob {
...
-            // never `deinit()`. `borrowed_view()` is the sound Rust spelling: it
-            // clones the `StoreRef`/`name` (whose `Drop`s balance the +1) and
-            // aliases `content_type`; `dupe()` would leak the boxed
-            // `content_type` copy. `as_class_ref` is the safe shared-borrow
-            // downcast — the JS wrapper roots the payload while `arg` is on the
-            // stack.
+            // never `deinit()`. `borrowed_view()` is the sound Rust spelling: a
+            // self-owning `dupe()` whose `StoreRef`/`name`/`content_type` are
+            // all released by drop glue at scope exit. `as_class_ref` is the
+            // safe shared-borrow downcast — the JS wrapper roots the payload
```
[the remark this answers is not in the document]
Comment (Jarred-Sumner) [reply src/runtime/node/types.rs]: Fixed in d3ded9c4a4e0 — the `Self::Blob` arm's comment now describes the actual ownership (a self-owning `dupe()` whose store/name/content_type are all released by the `Box<Blob>` drop glue) and notes that the explicit `store.take()` just mirrors Zig's teardown order, with the field drop seeing `None` afterwards. Kept the `take()` itself to preserve the explicit teardown-order pattern used at the other deinit sites.

File: src/jsc/webcore_types.rs [hunk 3]
```diff
@@ -486,6 +474,21 @@ impl Blob {
...
+/// every consumer that drops a `dupe()`d / `use_()`d / deserialized `Blob`
+/// without an explicit `deinit()` release the heap-allocated content type.
+///
+/// `free_content_type` only frees when `content_type_allocated` is set and
+/// resets the flag, so the existing explicit `deinit()` / `free_content_type()`
+/// call sites layered under this `Drop` cannot double-free.
+impl Drop for Blob {
+    fn drop(&mut self) {
+        self.free_content_type();
+    }
```
[the remark this answers is not in the document]
Comment (Jarred-Sumner) [reply src/jsc/webcore_types.rs:458]: Done — both comments now say `Blob::drop` frees `content_type` when `*self` is replaced and that the explicit call is kept to mirror Zig's teardown order (idempotent), matching the wording at the other updated sites. The explicit calls themselves are kept, per the second option in the suggestion.

[the remark this answers is not in the document]
Comment (Jarred-Sumner) [reply src/jsc/webcore_types.rs:458]: Done in a160948a7cde — both named sites plus a full sweep of the worktree for any remaining comment describing the pre-Drop ownership model.

Updated (5 sites): Blob.rs:5233 (dropped the "since Blob has no Drop" clause, kept the take-the-value rationale), Body.rs:845 (the scopeguard deinit rationale), Body.rs:1128 and Body.rs:1870 (the two "`Any` has no Drop … detach() reclaims it" BodyMixin arms), ByteBlobLoader.rs:257 (the same pattern in `to_buffered_value`). All now use the established "`Blob`'s drop glue frees it; the explicit call mirrors Zig's teardown order and is idempotent" wording.

Checked and deliberately left alone: Body.rs:2490 ("Zig has no Drop" refers to the Zig language, about `strong::Optional`), Body.rs:663 (ValueError refcounts), fetch.rs:1561/1572 (about `HTTPRequestBody` having no Drop impl and the Store ref — a different type and a different resource, unchanged by this PR), StaticRoute.rs:80 (already correctly states AnyBlob has drop glue), and the 10+ sites already updated in faa8eaae/d3ded9c4/b9f586db. A grep for no-Drop / drop-glue / would-leak / not-covered phrasing near Blob/content_type returns no remaining stale claims.

**Review by Jarred-Sumner — COMMENTED** [review]
Remove all code commetns. Make sure we're not cloning the content_type when it's not needed (i.e. it should remain interned for the static ones)

File: src/jsc/webcore_types.rs [hunk 4]
```diff
@@ -467,11 +436,8 @@ impl Blob {
...
 
+    /// `Blob.deinit()` (Blob.zig:3720). Tear down owned resources; if
     /// heap-allocated, also frees the heap box.
-    ///
-    /// PORT NOTE: kept as an explicit method (not `Drop`) because `Blob` is the
-    /// `m_ctx` payload of a `.classes.ts` class — `finalize()` owns teardown,
-    /// and many call sites tear down stack copies explicitly.
     pub fn deinit(&mut self) {
         self.detach();
         self.name.set(bun_core::String::dead());
```
Comment (Jarred-Sumner) [inline src/jsc/webcore_types.rs:443]: Does this leak the name property? Is it deref'ing the name?

Comment (Jarred-Sumner) [reply src/jsc/webcore_types.rs:443]: No leak — `name` is an `OwnedStringCell`, and `OwnedStringCell::set` is `self.0.replace(new).deref()` (src/bun_core/string/mod.rs:2756-2758), so the old name is deref'd when we overwrite it with `dead()`. That matches the Zig deinit (`this.name.deref(); this.name = .dead;`, Blob.zig:3722-3723). The dupe path is balanced too: `self.name.clone()` does `dupe_ref()` and the cell's `Drop`/`set` derefs it. The type's doc comment exists precisely because plain `Cell<String>::set` used to leak the old +1; this field was already migrated.
